code/modules/events/time_traveler.dm

```diff
@@ -0,0 +1,70 @@
+/datum/round_event_control/time_traveler
+	name = "Time Traveler"
+	typepath = /datum/round_event/ghost_role/time_traveler
+	weight = 15
+	max_occurrences = 1
+
+/datum/round_event/ghost_role/time_traveler
+	minimum_required = 1
+	role_name = "time traveler"
+	fakeable = TRUE
+
+/datum/outfit/time_traveler
+	uniform = /obj/item/clothing/under/pants/youngfolksjeans
+	suit = /obj/item/clothing/suit/jacket/letterman
+	belt = /obj/item/storage/belt/fannypack/blue
+	gloves = /obj/item/clothing/gloves/fingerless
+	shoes = /obj/item/clothing/shoes/wheelys
+	mask = /obj/item/clothing/mask/cigarette
+	glasses = /obj/item/clothing/glasses/orange
+	r_hand = /obj/item/computer_mouse
+
+/datum/round_event/ghost_role/time_traveler/proc/createEffects(var/turf/T)
+	playsound(T, 'sound/magic/lightningbolt.ogg', 50, 0)
+	var/obj/effect/portal/P = new /obj/effect/portal(T)
+	P.color = "#00ff00"
+	sleep(100)
+	qdel(P)
+
+/datum/round_event/ghost_role/time_traveler/announce(fake)
+	priority_announce("Bluespace-time distortions have been detected aboard the station.")
+/datum/round_event/ghost_role/time_traveler/spawn_role()
+	var/list/candidates = get_candidates(ROLE_ALIEN, null, ROLE_ALIEN)
+	if(!candidates.len)
+		return NOT_ENOUGH_PLAYERS
+
+	var/mob/dead/selected = pick(candidates)
+
+	var/datum/mind/player_mind = new /datum/mind(selected.key)
+	player_mind.active = TRUE
+
+	var/list/spawn_locs = list()
+	var/area = pick(1,3)
+	switch(area)
```

Comment:
Can't you use a landmark here? You're looping through every area in the world, and then every turf of those areas

Reply:
what

Reply:
can't you
use
a
landmark
placed
on
the
map,
like, the xeno spawn one,
because,
this
is
wasteful

do 
you
understand?